syntax = "proto3";

package service1;

service Products {
  rpc CreateProduct (Product) returns (Product);
  rpc GetProduct (ProductId) returns (Product);
  rpc ListProducts (Empty) returns (Products);
}

message Product {
  string id = 1;
  string title = 2;
  int32 passenger_capacity = 3;
  int32 maximum_speed = 4;
  bool in_stock = 5;
}

message ProductId {
  string id = 1;
}

message Products {
  repeated Product products = 1;
}

message Empty {}
